### DISCUSSION TITLE
APOE: successful regression run

### DISCUSSION THREAD

**Discussion by krootesmurdy** [trendscenter/coinstac#1506] [General]
Our three sites across three machines (two on GSU networks) were able to run vbm regression yesterday and get [results][(url)](https://drive.google.com/file/d/1o6cXTisDTv14O1prZGADQzhyJNwEU8-K/view?usp=sharing) (trying to replicate success again today).

**Comment by krootesmurdy** [comment]
successful run again with two sites on two machines. Here are the changes:

1. CLEAR LOCAL DATA after every run (even if error out)
2. NO auto-map (I think this was a major issue yesterday. the auto-map section *sometimes* adds or deletes a column (maybe from a previous run). I had to map the columns myself. Sandeep documented this problem yesterday too.
3. must delete any participant in the csv file with any missing data (regardless of if that column is used)

**Reply by dylanmartin** [reply]
For #3 is the expected behavior that the computation just ignores rows with missing data?

**Reply by krootesmurdy** [reply]
@spanta28 we discussed the missing data but I think we determined that it would be difficult to add that code in? Is that correct? So we opted to just have all the missing data removed.

**Reply by dylanmartin** [reply]
When you say missing data, do you mean missing files?

**Reply by krootesmurdy** [reply]
not missing files, missing covariate information (e.g., no information on site or sex).

**Comment by spanta28** [comment]
When any of the data is missing from the covariate.csv files , we could
have the code ignore missing values and perform regression but from data
science perspective we would also need to show the consortia members how
much data did the original files have and how much data was actually used
in the regression so that it presents the full picture. Or I just thought
of this we could just throw a message and say missing data will be ignored
and leave it there and have code just ignore missing values.

Anyways at the time we did not want to delay the process so I just asked
Kelly that the sites remove missing data. Not sure if that’s going to be
followed by everyone only an experiment will tell. But if the error points
to missing values then having a one liner code to ignore missing values is
easy enough to put in.

**Reply by dylanmartin** [reply]
Does this handling of rows with missing cells apply to all computations or are there some computations that can deal with it?
I agree we should give some kind of instruction to the user so that they know what the problem is and what data we're using if we come up with our own way around it.

**Reply by dylanmartin** [reply]
It seems like this lies firmly in the realm of data cleaning and preparation. I don't think we should go down the path of taking responsibility of cleaning the user's data. We should just give them guidelines for how it should look when consumed by COINSTAC. Perhaps we could put effort into making tools and guides for them to prepare data but I don't think we should sprinkle those responsibilities throughout the app. If we do handle data prep in the app, it should be done in an intentional way with the code organized and centralized maybe in a module or something.

**Reply by krootesmurdy** [reply]
To Sandeep's point, if we could have a message state that the missing data is noted and will be ignored, that would be great!!

**Reply by dylanmartin** [reply]
Agreed, that would be great. It may or may not be easy depending on how the .csv is consumed. @rssk do computations read the raw .csv or is there an intermediate state in COINSTAC where we could prune the bad rows?

**Reply by rssk** [reply]
Coinstac consumes the csv and gives the data to the computation after doing some type casting, I'm not sure I agree on the data formatting bit, the format is simple and should be very easy to check. @krootesmurdy Im unsure what exactly #3 is detailing, can you give an example?

**Reply by krootesmurdy** [reply]
for #3, if someone is missing a covariate (e.g., sex) then the blank cell causes issues. I believe the error is the X/Y mismatch but I mentioned I can confirm that tomorrow.

**Comment by dylanmartin** [comment]
@krootesmurdy what happens when you don't do step 3?
Does the UI crash?
Does the computation throw an error?
Does the pipeline run successfully?

**Reply by krootesmurdy** [reply]
I believe it's the X/Y mismatch error but I can double check tomorrow. 

We determined this problem a few weeks back, it's not related to running multiple sites.

**Comment by dylanmartin** [comment]
I just realized that some runs don't use all of the available covariates. Meaning the process of removing rows should know about which covariates are needed in case there are useful rows with a single blank cell for an unused covariate.

**Reply by krootesmurdy** [reply]
exactly, removing the rows means we lose data that may be helpful for other analyses.

**Reply by rssk** [reply]
just to put it here, here is the proposed solution
https://github.com/trendscenter/coinstac/issues/1519

**Reply by rssk** [reply]
This would remove rows if the current selected covar is empty for that subj

**Reply by krootesmurdy** [reply]
oh, that's exactly what we need!

**Reply by dylanmartin** [reply]
Here's the PR for the solution:
https://github.com/trendscenter/coinstac/pull/1522

![image](https://user-images.githubusercontent.com/7193853/194387839-fbf2637f-205e-466e-9b37-5c67c493a8b9.png)
